### PR TITLE
refactor perf script to separate load from exec

### PR DESCRIPTION
**Pull request by iritkatriel**

### CODE REVIEW

**Comment by gvanrossum** [comment]
Can you explain the effect this has? (IIUC the effect is that the lazy steady-state is no longer significantly slower?)

**Comment by iritkatriel** [comment]
I don't see an effect on lazy steady state, I see that classic steady state becomes as slow as lazy steady state if we do lazy first and classic second. 

The refactor of this PR separates the load from the exec stages, I wanted to see whether doing just lazy-load (but no lazy-exec) before classic has the same impact on classic steady-state, and it does.  So we need to find what is it that lazy load is doing to mess things up for exec (I'm assuming that classic and lazy exec are both getting messed up in the same way) .

So it's maybe leaking a lot of small objects in cycles, making GC slower? Or trashing the memory somehow?

File: Tools/scripts/lazy_marshal_perf.py [hunk 1]
```diff
@@ -35,14 +37,27 @@ def helper(data, label, lazy):
...
-    new_timings = helper(data, "Lazy", lazy=True)
+    # new_timings = helper(data, "Lazy", new_timings, lazy=True)
+    # classic_timings = helper(data, "Classic", classic_timings, lazy=False)
+
+    new_timings = {}
+    classic_timings = {}
+    classic_codes = load_helper(data, "Classic", classic_timings, lazy=False)
+    lazy_codes = load_helper(data, "Lazy", new_timings, lazy=True)
+    exec_helper(classic_codes, "Classic", classic_timings)
+    exec_helper(lazy_codes, "Lazy", new_timings)
```
Comment (iritkatriel) [inline Tools/scripts/lazy_marshal_perf.py:60]: So the idea is to play around with the order of these 4 lines and see the impact.

**Comment by gvanrossum** [comment]
Wonder if the timings become much less stable if you reduce from 1000 to
100 iterations on line 23? That would make experimentation much quicker. :-)

**Comment by iritkatriel** [comment]
With 100 iterations the numbers are very close.

**Comment by gvanrossum** [comment]
Let me just land this. I'll add the ns.clear() thing separately.

**Comment by iritkatriel** [comment]
ok, we'll have to see in real benchmarks what happens.

It was a good day - started with a total mystery and now everything is clear.
See you tomorrow.

**Pull request merged**